persistence-modules/spring-boot-persistence-2/src/test/java/com/baeldung/db/indexing/FileSystemImageIntegrationTest.java

```diff
@@ -0,0 +1,58 @@
+package com.baeldung.db.indexing;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.io.InputStream;
+
+import org.assertj.core.api.Assertions;
+import org.junit.FixMethodOrder;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMultipartHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import com.baeldung.db.indexing.ImageUploaderApplication;
+
+@SpringBootTest(classes = ImageUploaderApplication.class)
+@AutoConfigureMockMvc
+@FixMethodOrder
+class FileSystemImageIntegrationTest {
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @Test
+    @Order(1)
```

Comment:
Bad practice to use ordered tests - if a test case requires something from earlier, then it's probably one test case.

Reply:
I find this topic discussable. I sure understand your point. But I see a trade-off here. If you consider that your tests should be as clean as possible, we should follow the Single Responsibility Principle. And asserting two different flows (save and find) is a clear violation of it. 
Sure, if we create complex tests relying on order, it may become a mess. But the same is valid for huge test methods with many assertions.

With the order approach, we also have the benefit of a strict view of where is the problem, without depending on logs.

Anyway, I removed the ordering but kept both methods. Now the download test is relying on a mock. Kind of a middle ground. 

As an integration test, I still see more ups than downs in ordering our tests. For these specific cases.